ArcGISRuntimeSDKQt_CppSamples/AR/DisplayScenesInTabletopAR/DisplayScenesInTabletopAR.cpp

```diff
@@ -150,7 +151,17 @@ void DisplayScenesInTabletopAR::createScenePackage(const QString& path)
 {
   m_scenePackage = new MobileScenePackage(path, this);
   connect(m_scenePackage, &MobileScenePackage::doneLoading, this, &DisplayScenesInTabletopAR::packageLoaded);
-  m_scenePackage->load();
+
+  connect(m_permissionsHelper, &PermissionsHelper::requestFilesystemAccessCompleted, this, [this]()
+  {
+    m_scenePackage->load();
```

Comment:
How much error handling do we want in the samples? AR can be complicated, so perhaps it's best to assume we are granted permissions and let the load fail for the scene package if we're wrong.

The other option is to add the error handling logic:
```c++
if (m_permissionsHelper->fileSystemAccessGranted())
  m_scenePackage->load();
else
  qDebug() << "some error message here"; // do we want to pop up an error dialog?
```

Users won't see the error message unless they're running from Qt Creator and watching the console, so that's probably not enough, either.

@ldanzinger your thoughts? I'm leaning towards keeping the sample code as clean and tight as possible. We can possibly mention in the readme that filesystem permissions are required and leave it at that.

Reply:
Please correct me if I am wrong, but are you proposing that we don't request filesystem permissions at all and just let the mobile scene package fail to load unless the user manually allows access through the settings? Wouldn't this remove the need for the PermissionsHelper code all together?

Reply:
>but are you proposing that we don't request filesystem permissions at all

No, we have a misunderstanding. There are two steps we need to consider here:
1. The sample code must allow the app to request permissions. Permissions Helper does this.
2. The user has to actually click "accept" or "ok" on the permissions dialog.

What I'm talking about is if in step 2, the user clicks "deny", either on purpose or on accident. Then what?  The type of graceful error handling code for that situation could add lots of extra logic to provide a good UX.  The main point is that this sample is for showcasing AR with Qt workflows and not handling permissions on Android.

Reply:
That makes complete sense.

I think adding to the readme would be ideal. Currently it checks to see if permissions were granted. If they have not, it will prompt every time the app opens unless they check "Don't ask again". So the readme would be there in the rare case they check the box and hit deny which I think should suffice.

Reply:
@DonMoJa cool, that makes sense to me as well. `requestFilesystemAccessCompleted` lambda is good as-is then.

Reply:
Agreed on all this. We need to make the sample functional but not super robust to handle all scenarios, since this isn't the main purpose of the sample. Adding comments or readme details will be sufficient